app/src/main/java/pm/gnosis/heimdall/ui/transactions/view/review/ReviewTransactionActivity.kt

```diff
@@ -167,6 +168,18 @@ class ReviewTransactionActivity : ViewModelActivity<ReviewTransactionContract>()
         }
     }
 
+    override fun onConfirmationDialogDismiss() {
+        // If we have a reference id then we have been opened from a external request and should just close the screen without opening a new one
+        if (referenceId ?: 0 < 0)
```

Comment:
shouldn't we have an else? also why `< 0`?

Reply:
default value of referenceId should be -1. referenceId ?: -1 < 0
Else case here should finish current activity.